### ISSUE TITLE
Reduce sleep comfort if liquid or fields are on the tile you are sleeping

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#59047]
### Is your feature request related to a problem? Please describe.

That should make sleeping in a sort-of clean environment more desirable.

### Solution you would like.

1. If e.g a puddle of blood or a heap of gore or vomit is on your bed, reduce the comfort you get from sleeping and prohibit getting the well rested morale boost. Also apply a morale debuff because it is gross.
2. The outdoorsman trait could make it easier to endure.
3. Squeamish should prohibit sleeping in such conditions.
4. Add a chance of an outer clothing part getting filthy when sleeping in a puddle of blood or heap of gore. Could also apply when being prone in such a field in general.

Basically account for any spilled liquid. I currently can not think of any that would be neutral.

The fields on the other hand are a bit more tricky. You should already see what I am aiming for. If there is just a blood splatter or these twigs and leaves from triffids it should be mostly fine. But a heap of gore, a puddle of blood and/or bile?
"You find it gross and really hard to sleep in this yucky place".

Fog could be fine for the player, as would be tobacco smoke for smokers. Relaxation gas should not affect sleep in a negative way at all.

**Comment by AtomicFox556** [comment]
I think it'd still be a reason to clean up only the tile that you're sleeping on, but I guess it beats completely ignoring it. But then again, any bed that is thoroughly covered in blood would need much more cleaning than just wiping it with a rag.

As a related feature, you could also have a chance for outermost footwear to become filthy as you walk through any gore; it'd actually be at least some kind of incentive to clean up everything rather than just the bed.

Also I guess psychopath characters could well ignore the effects of trying to sleep in a pile of gore.

**Comment by PatrikLundell** [comment]
Not sure about the psychopath. They don't care about the person the gore came from, but the gore itself is sticky, stinky, wet, and generally unpleasant for the psycho itself.
